src/Test/Utilities/Shared/Compilation/CompilationExtensions.cs

```diff
@@ -148,5 +150,102 @@ internal static void VerifyAssemblyAliases(this Compilation compilation, params
+            {
+                if (declaration.DeclaredNode == null ||
+                    declaration.DeclaredSymbol == null || 
+                    !CanHaveExecutableCodeBlock(declaration.DeclaredSymbol))
+                {
+                    continue;
+                }
+
+                if (skipImplicitlyDeclaredSymbols && declaration.DeclaredSymbol.IsImplicitlyDeclared)
+                {
+                    continue;
+                }
+
+                if (!string.IsNullOrEmpty(symbolToVerify) && !declaration.DeclaredSymbol.Name.Equals(symbolToVerify, StringComparison.Ordinal))
+                {
+                    continue;
+                }
+
+                actualTextBuilder.Append(declaration.DeclaredSymbol.ToDisplayString());
+                if (declaration.ExecutableCodeBlocks.Length != 1)
+                {
+                    actualTextBuilder.Append($" ('{declaration.ExecutableCodeBlocks.Length}' executable code blocks)");
+                }
+                
+                foreach (var executableCodeBlock in declaration.ExecutableCodeBlocks)
+                {
+                    actualTextBuilder.Append(Environment.NewLine);
+                    var operation = model.GetOperation(executableCodeBlock);
+                    if (operation != null)
+                    {
+                        var operationTree = OperationTreeVerifier.GetOperationTree(operation, initialIndent: 2);
+                        actualTextBuilder.Append(operationTree);
+                    }
+                    else
+                    {
+                        actualTextBuilder.Append($"  SemanticModel.GetOperation() returned NULL for code block with kind '{executableCodeBlock.GetKind()}'");
+                    }
+                }               
+
+                actualTextBuilder.Append(Environment.NewLine);
+            }
+
+            var assertFailed = false;
+            var newLineChars = Environment.NewLine.ToCharArray();
+            var actual = actualTextBuilder.ToString().Trim(newLineChars);
+            expectedOperationTree = expectedOperationTree.Trim(newLineChars);
+
+            try
+            {
+                Assert.Equal(expectedOperationTree, actual);
+            }
+            catch (EqualException)
+            {
+                assertFailed = true;
+                throw;
+            }
+            finally
+            {
+                if (assertFailed)
+                {
+                    Console.WriteLine($"Actual operation tree:\r\n{actual}\r\n");
+                }
+            }
+        }
+
+        internal static bool CanHaveExecutableCodeBlock(ISymbol symbol)
+        {
+            switch (symbol.Kind)
+            {
+                case SymbolKind.Field:
+                case SymbolKind.Event:
+                case SymbolKind.Method:
+                case SymbolKind.NamedType:
```

Comment:
This matches http://source.roslyn.io/Microsoft.CodeAnalysis/R/0e1dc66f2235f48c.html